Dep react

notAro14 opened the pull request:
@jadynekena si tu veux regarder un peu comment c'est fait.

**Ce n'est pas encore déployable tel quel**. C'est plus un premier jet pour l'instant, démontrant surtout la vue table des données.

J'ai mis dans le `README` les instructions pour lancer l'application. J'ai déplacé tout le code vanilla JS dans le dossier `legacy-dist`.

Commit:
Setup and install Next.js

Commit:
Add constants and setup env variables

Commit:
Implement data fetching

Commit:
Change data types

Commit:
Display data with react table and chakra ui

Commit:
Refactor components

Commit:
Specify spinner role and add export script

Commit:
Format README

Commit:
Update README

To include React app development setup

Commit:
Rename vanilla folder to legacy-dist

Commit:
Copy static files to legacy-dist

notAro14 commented:
Voici un aperçu de l'app actuelle avec React

<img width="1337" alt="Screenshot 2022-08-29 at 21 59 32" src="https://user-images.githubusercontent.com/47597252/187287678-9ee7746d-ba83-4568-8066-72bd318585c8.png">

Commit:
Add filtering and improve sorting ui styles

Commit:
Improve visa requirements label

Commit:
Add details modal

Commit:
Update visa text colors

jadynekena commented:
@notAro14 
déjà merci pour la contribution :)

J'ai pu tester en local (avec npm d'ailleurs ça suffisait), et du coup si j'ai bien compris, cette partie concerne uniquement l'affichage brut des données sous forme de table (donc ça serait le lien malagasypassport.ml/src ?).

De ce que j'ai vu ça marche plutôt bien. La recherche et le tri c'est vraiment top !

Je te note ci-dessous les points importants que j'ai constatés.

## Les colonnes
C'est possible de garder toutes les colonnes sur le tableau, à part le country code ? 
![image](https://user-images.githubusercontent.com/26454443/187398793-f694d870-5075-40a7-ae34-bc3b7346dd6f.png)
Parce qu'en tant que voyageur, les gens voudront par exemple pouvoir chercher par continent, durée possible, etc. Et surtout vérifier la source des affirmations sur le visa, et voir si c'est avéré par un témoignage. Avec un autre contributeur on va bientôt remplir "shortest itinerary example" aussi, ce serait bien de garder cette info.   
Du coup : tout garder sauf les **_GSHEET_ROW_NUMBER** (par défaut envoyé par la source) et **Country code**.

## Le format
1. Si on reste sur l'idée de tout "cacher" derrière un (i) à part le country et le visa requirements (et continent à rajouter si possible), sur un desktop je pense que ce serait + lisible si tout était compacté au centre parce qu'il n'y a que 2-3 colonnes affichées au final : 
![image](https://user-images.githubusercontent.com/26454443/187401965-53050545-be77-408f-ae59-df846c509c56.png)
L'espacement initial entre les 2 colonnes était assez conséquent, c'était un peu dur d'identifier le visa requirement d'un pays

 2. Si on garde toutes les colonnes comme suggéré dans le paragraphe **Les colonnes**, alors on peut garder ton format actuel.

## Les données
A chaque clic, j'ai constaté que les données sont re-récupérées via le lien d'API:
![image](https://user-images.githubusercontent.com/26454443/187403765-58bfb87b-b081-4164-9ce1-ecabfba713eb.png)
En prod ca va être impossible à faire, aujourd'hui il y a en moyenne 3000clics par jour, donc si on sollicite 3000x le serveur va perdre les boules lol
![image](https://user-images.githubusercontent.com/26454443/187405362-9d53b2ff-d410-4f26-877a-11e72ed400fb.png)

D'où la solution de : fetch uniquement au chargement, et utiliser sessionStorage pour manipuler les données pendant la session.



## Les infos
Je suppose que le contenu derrière ce (i) n'est pas terminé ?
![image](https://user-images.githubusercontent.com/26454443/187400244-9ad91c82-f7b7-4a7b-9706-9cf3dc130d7b.png)
![image](https://user-images.githubusercontent.com/26454443/187400283-6d47f970-4a40-4c07-a28e-050316de4318.png)


En tout cas top, on prendra le temps d'en discuter en détails si t'as un moment !

Commit:
Replace pnpm by npm

Commit:
Add link to source column

And add tooltip to see details button

Commit:
Add datalist on filters and add continent column

Commit:
Add verified status column

Commit:
Add maxLengthStay column

Commit:
Improve overall UI and UX

- add more information in details modal
- make table responsive (by adding TableContainer)
- reorganize column order

Commit:
Update README

notAro14 commented:
@jadynekena J'ai apporté qlq modifications

- utilisation de `npm` à la place de `pnpm` (étant donné que la majorité des gens utilisent `npm`)
- ré organisation des colonnes pour faciliter la lisibilité
- résolution d'un petit bug qui ne permettait pas de combiner les résultats des filtres
- finition de la modale **Details**
- ajout d'autres colonnes

![Screenshot 2022-08-30 at 20 38 29](https://user-images.githubusercontent.com/47597252/187516862-e37979d5-2da6-4941-8415-b991291d6fa0.png)

![Screenshot 2022-08-30 at 20 38 41](https://user-images.githubusercontent.com/47597252/187516888-e852d0e9-46fe-4fef-82f8-2c1a924bb92d.png)

PS : ma prochaine réflexion sera sur la persistence des données pour éviter d'envoyer trop de requêtes au serveur
